### PR TITLE
Oc 178 profiles

### PR DESCRIPTION
**Pull request by ZaraGlynn**

### CODE REVIEW

File: src/profiles/profiles.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  try {
+    const
+      data = JSON.parse(event.body);
+
+    await Joi.validate(data, Joi.object().keys(
+      {
+        id: Joi.number().integer().required(),
+        profile_image: Joi.string(),
+        featured_image: Joi.string(),
+        full_name: Joi.string().required(),
```
Comment (matthewberryman) [inline src/profiles/profiles.ts]: not sure why we need to have full_name required on update?
also we should check that the cognito user is an admin, or if not then that their uuid matches one of the uuids associated with the profile

**Comment by matthewberryman** [comment]
In terms of search, I don't think we need display more than name and institution? And only where public_profile = true

File: src/profiles/profiles.ts [hunk 2]
```diff
@@ -145,3 +163,40 @@ export const update = async (event: APIGatewayProxyEvent): Promise<APIGatewayPro
...
+    await Joi.validate(event.queryStringParameters, Joi.object().keys({
+      query: Joi.string().required()
+    }));
+
+    const
+      { query } = event.queryStringParameters,
+      params = [query],
+      sqlStatement = `
+        SELECT 
+          profiles.id,
```
Comment (Dan-Wood) [inline src/profiles/profiles.ts]: Why don't we grab everything?

File: src/profiles/profiles.ts [hunk 3]
```diff
@@ -145,3 +163,40 @@ export const update = async (event: APIGatewayProxyEvent): Promise<APIGatewayPro
...
+          profiles.full_name,
+          profiles.profile_type,
+          profiles.affiliation
+        FROM ${process.env.PROFILES_TABLE}
+        WHERE 
+          LOWER(full_name) LIKE LOWER($1) || '%'
```
Comment (Dan-Wood) [inline src/profiles/profiles.ts]: Only searching full name?

What about

field_expertise
city
country
position
contact_email
contact_person

I want to see every profile who is a "Whale Inspector"

File: src/profiles/profiles.ts [hunk 4]
```diff
@@ -145,3 +163,40 @@ export const update = async (event: APIGatewayProxyEvent): Promise<APIGatewayPro
...
+        AND public_profile = true
+      `;
+    const result = await db.any(sqlStatement, params);
+
+    return successResponse({ profile: result ? result : [] });
```
Comment (Dan-Wood) [inline src/profiles/profiles.ts]: Plural as you'd return more than one.

File: src/profiles/profiles.ts (hunk 2 above)
Comment (ZaraGlynn) [reply src/profiles/profiles.ts]: Per Matthews comment
>In terms of search, I don't think we need display more than name and institution? And only where public_profile = true

Institution is a profile type so we get affiliation instead and only return their name and profile type

File: src/profiles/profiles.ts (hunk 3 above)
Comment (Dan-Wood) [reply src/profiles/profiles.ts]: @ZaraGlynn is this resolved?

**Comment by Dan-Wood** [comment]
Should we split “full name” into first and last? @matthewberryman @ZaraGlynn 

There doesn’t seem to be any security on update? Anyone could make the request with a random ID and change the that ID's name to Matthew from Dan.

Update should only update if the id matches the incoming UUID (I suggest we get this one from Cognito, see Mailchimp as we don't want a user to fake the UUID either.) 

That said, we're not storing the UUID on the profile so there's no way to match two and two together so how do I update my details from the UI, Cognito doesn't know my database ID so I can't get it from there and the database doesn't hold my UUID. Or am I missing something?

And Admin Update is obviously not constrained at all.

**Comment by matthewberryman** [comment]
Yes do need to match uuid from cognito as discussed.

Some people don't have separate first and last names, and which one comes first (given or family name) depends on language. I think just keep full name and search on substrings.

**Pull request merged**